### PR TITLE
Convert the generate-octicons script to TypeScript

### PR DESCRIPTION
**Pull request by j-f1**
Also, update the `fs` calls in the script to use UTF-8 when reading and writing.

### CODE REVIEW

**Comment by j-f1** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/20013

**Comment by j-f1** [comment]
🏆

File: script/generate-octicons.ts [hunk 1]
```diff
@@ -9,11 +7,11 @@
 
 'use strict'
```
Comment (shiftkey) [inline script/generate-octicons.ts]: :fire: this line too

File: script/generate-octicons.ts [hunk 2]
```diff
@@ -24,12 +22,23 @@ const filePath = path.resolve(
   'sprite.octicons.svg'
 )
 
-const file = fs.readFileSync(filePath)
+const file = fs.readFileSync(filePath, 'utf-8')
```
Comment (shiftkey) [inline script/generate-octicons.ts]: we need to add a header to disable the warning about sync methods:

```ts
/* tslint:disable:no-sync-functions */
```

File: script/generate-octicons.ts [hunk 3]
```diff
@@ -24,12 +22,23 @@ const filePath = path.resolve(
...
+
+interface XML2JSResult {
```
Comment (shiftkey) [inline script/generate-octicons.ts]: There's a tslint rule about interfaces starting with an `I` that is adding noise in here. Not sure whether this is also enforced by `eslint`, but we should add that back until we can do the transition.

File: script/generate-octicons.ts [hunk 4]
```diff
@@ -24,12 +22,23 @@ const filePath = path.resolve(
...
+  svg: { symbol: ReadonlyArray<XML2JSNode> }
+}
+interface XML2JSNode {
```
Comment (shiftkey) [inline script/generate-octicons.ts]: Same thing as above

File: script/generate-octicons.ts [hunk 5]
```diff
@@ -9,11 +7,11 @@
 
 'use strict'
 
-const fs = require('fs')
-const process = require('process')
-const xml2js = require('xml2js')
-const path = require('path')
-const toCamelCase = require('to-camel-case')
+import fs = require('fs')
+import process = require('process')
```
Comment (shiftkey) [inline script/generate-octicons.ts]: And this can be removed - `process` is a global and only used for `process.exit` here.

**Comment by j-f1** [comment]
🔭

**Pull request merged**